Book-A-Desk.Api/BookingsHttpHandler.fs

```diff
@@ -99,68 +99,53 @@ module BookingsHttpHandler =
                     return! json response.Error next context
             }
 
-        let handleGetByEmailAndDateFromEventStore eventStore email date = asyncResult {
+        let handleGet eventStore date officeId email = asyncResult {
             let! bookingEvents = eventStore.GetEvents ()
-            return! ReservationsQueriesHandler.getUserBookingsStartFrom bookingEvents email date
-        }
-        
-        let handleGetByDateFromEventStore eventStore date = asyncResult {
-            let! bookingEvents = eventStore.GetEvents ()
-            return! ReservationsQueriesHandler.getUsersBookingsStartFrom bookingEvents date
+
+            match officeId with
```

Comment:
So, I see why you have changed this code to sometimes call `getFilteredBookings` and sometimes call `getUserBookingsStartFrom` (failing api tests). These tests were implemented when we partially implemented the "Upcoming Bookings" feature. For now, let's ignore this "Upcoming Bookings" feature, we will revisit it later.

For now, I think you can do this:
1. We will reimplement "Upcoming Bookings" later, so all logic relating to it can be removed. In this case, you can change this `handleGet` to only call `ReservationsQueriesHandler.getFilteredBookings`
2. The API tests will break, however what we will do is to change them to:
    - Remove all tests that talk about "returning bookings of future date".
    - Add new tests to test our filtering (keep them at a high level. Our unit tests are responsible for testing all of the combinations, these tests should test only 1-2 combinations)

When we actually reimplement "Upcoming Bookings", we will add another query parameter (let's call it startFrom for now) to specifically have bookings starting from a certain date

Reply:
Another note:
If we ever run into the same situation, where some parameters mean we should call different logic, this check should be done in the Domain. The Domain should handle all logic, the Api Handlers should only do two things:
1. Call the Infrastructure code (to get the Booking Events)
2. Call the Domain
3. Transform types to and from the API definition and handle API-related concepts like error codes.